src/lib/storage/table.cpp

```diff
@@ -17,53 +17,84 @@
 
 namespace opossum {
 
+Table::Table(const uint32_t chunk_size) : _max_chunk_size(chunk_size) {
+  // On table creation, a first chunk shall be created.
+  _append_new_chunk();
+}
+
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  DebugAssert(row_count() == 0, "You can only add columns when no data has been added");
+
+  _column_names.push_back(name);
+  _column_types.push_back(type);
+  _chunks.back().add_segment(make_shared_by_data_type<BaseSegment, ValueSegment>(type));
 }
 
 void Table::append(std::vector<AllTypeVariant> values) {
-  // Implementation goes here
+  if (_chunks.back().size() >= _max_chunk_size) {
+    _append_new_chunk();
+  }
+
+  _chunks.back().append(values);
```

Comment:
Currently, the tests would pass even if we appended some garbage here (we only check that it gets bigger). I don't see a practicable way to unit-test this though, so I'd say we should leave the test as is now and maybe ask in the next seminar session about mock objects.